### ISSUE TITLE
`onToken` output does not match the parsed TypeScript program

### ISSUE THREAD

**Issue by magic-akari** [sveltejs/acorn-typescript#82]
## Reproduction

```js
import { Parser } from 'acorn';
import { tsPlugin } from '@sveltejs/acorn-typescript';

const tokens = [];

Parser.extend(tsPlugin({ jsx: true })).parse('const f = <T,>(x: T) => x;', {
  ecmaVersion: 'latest',
  sourceType: 'module',
  onToken: tokens
});

console.log(
  tokens.map((token) => [token.type.label, token.start, token.end])
);
```

## Expected

Each token is reported once, using the token type selected for the returned TypeScript AST. The type parameter at offsets 11–12 is reported as `name`.

## Actual

The callback first reports `jsxTagStart` and `jsxName` tokens for offsets 10–12, then reports the same source range again as TypeScript tokens. The token stream therefore contains extra tokens that do not match the returned AST.
